src/jit/emitxarch.cpp

```diff
@@ -4888,19 +4916,54 @@ void emitter::emitIns_AX_R(instruction ins, emitAttr attr, regNumber ireg, regNu
 }
 
 #if FEATURE_HW_INTRINSICS
-void emitter::emitIns_SIMD_R_R_R(instruction ins, regNumber reg, regNumber reg1, regNumber reg2, var_types simdtype)
+void emitter::emitIns_SIMD_R_R_R(instruction ins, regNumber reg, regNumber reg1, regNumber reg2, emitAttr attr)
+{
+    if (UseVEXEncoding())
+    {
+        emitIns_R_R_R(ins, attr, reg, reg1, reg2);
+    }
+    else
+    {
+        if (reg1 != reg)
+        {
+            emitIns_R_R(INS_movaps, attr, reg, reg1);
+        }
+        emitIns_R_R(ins, attr, reg, reg2);
+    }
+}
+
+void emitter::emitIns_SIMD_R_R_R_R(
+    instruction ins, regNumber reg, regNumber reg1, regNumber reg2, regNumber reg3, emitAttr attr)
 {
-    if (UseVEXEncoding() && reg1 != reg)
+    assert(ins == INS_vblendvps || ins == INS_vblendvpd || ins == INS_vpblendvb || ins == INS_blendvps ||
+           ins == INS_blendvpd || ins == INS_pblendvb);
+    if (UseVEXEncoding())
     {
-        emitIns_R_R_R(ins, emitTypeSize(simdtype), reg, reg1, reg2);
+        // convert SSE encoding of SSE4.1 instructions to VEX encoding
+        switch (ins)
```

Comment:
Is this the best or right place to handle this?

Reply:
Decided by the above limitation.

Reply:
Are these the only instructions that behave like that?